### PR TITLE
Add and fix more strict linting

### PR DESCRIPTION
**Pull request by fwiesel**
It is easier to do that early then later.

### CODE REVIEW

File: src/gerrit_argo_gateway/__init__.py [hunk 1]
```diff
@@ -29,48 +39,56 @@ class GerritGateway:
     # Example: myuser@yourhost:1234
     _SSH_RE = re.compile(r"(?P<username>[^@]+)@(?P<host>[^:]+)(?::(?P<port>\d+))?")
 
-    def __init__(self, subscriptions=None) -> None:
+    def __init__(self, subscriptions: list[str] | None = None) -> None:
+        """
+        Set up the gateway with filtered to the given subscriptions (or all if unset).
```
Comment (joker-at-work) [inline src/gerrit_argo_gateway/__init__.py]: typo: "with filtered to the given" - I think "with filtering"? not sure

File: src/gerrit_argo_gateway/__init__.py [hunk 2]
```diff
@@ -80,10 +98,11 @@ async def _trigger_build(self, event):
                 },
                 json=event,
             )
-            LOG.info(f"{event['type']}\t{event['project']}\t{event['changeKey']['id']}" f" -> {resp.status_code}")
+            LOG.info("%s\t%s\t%s -> %s", event["type"], event["project"], event["changeKey"]["id"], resp.status_code)
             resp.raise_for_status()
 
     async def __call__(self) -> None:
+        """Run the stream loop and only stops if either the connection drops or is closed via `stop()`."""
```
Comment (joker-at-work) [inline src/gerrit_argo_gateway/__init__.py]: typo: "run […] and only stops" should be either "runs […] and only stops" or "run […] and only stop"

File: src/gerrit_argo_gateway/__main__.py [hunk 3]
```diff
@@ -1,22 +1,27 @@
+"""
+Main module to be invoked via python -m gerrit_argo_gateway.
```
Comment (joker-at-work) [inline src/gerrit_argo_gateway/__main__.py:2]: nit: The `.` at the end of the line makes this harder to copy and paste. I'd get rid of it.

File: src/gerrit_argo_gateway/__main__.py [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
+
+Setups loggiging and the gateway and hooks up the signal handlers to stop the thread.
```
Comment (joker-at-work) [inline src/gerrit_argo_gateway/__main__.py]: typo: "loggiging" :D

Is "setups" the right word? I'd have chosen "sets up". I could not find "setup" as a verb in my online dictionary of choice.

File: tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+# noqa: D104
```
Comment (joker-at-work) [inline tests/__init__.py:1]: Maybe we can just put it in the default ignore list?

File: src/gerrit_argo_gateway/__init__.py (hunk 1 above)
Comment (fwiesel) [reply src/gerrit_argo_gateway/__init__.py]: I dropped the word "events", it seems.

Comment (fwiesel) [reply src/gerrit_argo_gateway/__init__.py]: Now
>  Set up the gateway for gerrit stream events filtered by subscriptions (or all if unset).

File: src/gerrit_argo_gateway/__main__.py (hunk 3 above)
Comment (fwiesel) [reply src/gerrit_argo_gateway/__main__.py:2]: It comes from the pydocstyle linter rule [D415](https://docs.astral.sh/ruff/rules/ends-in-punctuation/).

File: src/gerrit_argo_gateway/__main__.py (hunk 4 above)
Comment (fwiesel) [reply src/gerrit_argo_gateway/__main__.py]: Fixed

File: tests/__init__.py (hunk 5 above)
Comment (fwiesel) [reply tests/__init__.py:1]: Well, I think it is a good practice that all public modules should have a doc-string.
So `_whatever.py` wouldn't require a doc-string.

I find it only a bit stupid for `__init__.py`, and haven't found an option to exclude it there except putting the noqa rule there.

File: src/gerrit_argo_gateway/__init__.py (hunk 2 above)
Comment (fwiesel) [reply src/gerrit_argo_gateway/__init__.py]: fixed.

**Review by joker-at-work — APPROVED** [review]

**Pull request merged**